src/macros.rs

```diff
@@ -146,12 +144,10 @@ macro_rules! redis_module {
             ) } == raw::Status::Err as c_int { return raw::Status::Err as c_int; }
 
             let context = $crate::Context::new(ctx);
-            let args = $crate::decode_args(argv, argc);
-
-            if args.is_err() { return raw::Status::Err as c_int }
+            let args = $crate::decode_args(ctx, argv, argc);
 
             $(
-                if $init_func(&context, &args.unwrap()) == $crate::Status::Err {
+                if $init_func(&context, &args) == $crate::Status::Err {
```

Comment:
Change of order of potential failure due to none utf8 in args?
Used to return error before `init_func`.
Now could return the error when doing `next_redis_string` in the `init_func` (but some other logic could precede it in the `init_func`)
Do we care about such minor/rare behavior changes - I assume not.